published_date/published_date.py

```diff
@@ -0,0 +1,109 @@
+import logging, os
+import time
+import json
+from json.decoder import JSONDecodeError
+from requests_futures.sessions import FuturesSession
+from concurrent.futures import as_completed
+
+
+logger = logging.getLogger(__name__)
+
+
+class FetchPublishedDate:
+
+    def __init__(self, canvas_url, canvas_token, workers, canvas_ids):
+        self.canvas_url = canvas_url,
+        self.canvas_token = canvas_token,
+        self.canvas_ids = canvas_ids
+        self.workers = workers
```

Comment:
`num_workers`? `workers` makes me thing they are objects of some kind.

Reply:
Done